dataactbroker/handlers/fileHandler.py

```diff
@@ -411,48 +449,62 @@ def uploadFile(self):
             # Unexpected exception, this is a 500 server error
             return JsonResponse.error(e,StatusCode.INTERNAL_ERROR)
 
-    def generateD1File(self):
-        """ Initiates the generation of D1 """
-        jobDb = self.interfaces.jobDb
-        requestDict = RequestDictionary(self.request)
+    def startGenerationJob(self, submission_id, file_type):
+        """ Initiates the generation of D1
 
-        if not (requestDict.exists("submission_id") and requestDict.exists("start") and requestDict.exists("end")):
-            exc = ResponseException("Generate D1 Files route requires submission_id, start, and end", StatusCode.CLIENT_ERROR)
-            return JsonResponse.error(exc, exc.status)
+        Args:
+            submission_id: ID of submission to start job for
+            file_type: Type of file to be generated
 
-        submission_id = requestDict.getValue("submission_id")
-        start_date = requestDict.getValue("start")
-        end_date = requestDict.getValue("end")
+        Returns:
+            Tuple of boolean indicating successful start, and error response if False
 
-        if not (StringCleaner.isNumeric(submission_id) and StringCleaner.isDate(start_date) and StringCleaner.isDate(end_date)):
-            exc = ResponseException("submission id, start, and/or end cannot be parsed into their appropriate types", StatusCode.CLIENT_ERROR)
-            return JsonResponse.error(exc, exc.status)
+        """
+        jobDb = self.interfaces.jobDb
+        file_type_name = self.EXTERNAL_FILE_TYPE_MAP[file_type]
```

Comment:
I realize EXTERNAL_FILE_TYPE_MAP wasn't introduced in this PR, but for consistency does it make sense to load that dictionary from the file_type table (which would perhaps need an additional user-submitted vs generated flag)?